source/MaterialXView/Viewer.cpp

```diff
@@ -652,6 +668,44 @@ void Viewer::createAdvancedSettings(Widget* parent)
         _showAdvancedProperties = enable;
         updateDisplayedProperties();
     });
+
+    //Units
+    {
+        Widget* sampleGroup = new Widget(advancedPopup);
+        sampleGroup->setLayout(new ng::BoxLayout(ng::Orientation::Horizontal));
+        new ng::Label(sampleGroup, "Unit Space:");
+
+       
+        {
+            //TODO: Need a get supported units
+            mProcessEvents = false;
+            unitOptions.push_back("nanometer");
```

Comment:
To be driven by the supplied definitions, these can be taken from the length converter.